src/fsharp/fsc.fs

```diff
@@ -647,11 +647,13 @@ let main1OfAst
 
     let primaryAssembly =
         // temporary workaround until https://github.com/dotnet/fsharp/pull/8043 is merged:
-        // pick a primary assembly based on the current runtime.
+        // pick a primary assembly based on whether the developer included System>Runtime in the list of reference assemblies.
         // It's an ugly compromise used to avoid exposing primaryAssembly in the public api for this function.
-        let isNetCoreAppProcess = System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription.StartsWith ".NET Core"
-        if isNetCoreAppProcess then PrimaryAssembly.System_Runtime
-        else PrimaryAssembly.Mscorlib
+        let includesSystem_Runtime = dllReferences |> Seq.exists(fun f -> Path.GetFileName(f).Equals("system.runtime.dll",StringComparison.InvariantCultureIgnoreCase))
```

Comment:
@vzarytovskii ,
this is what fixed the il baseline issue. Here we set the primary  assembly differently on NET5.0 than netcoreapp3.1.
The change is to set the primary assembly to system_runtime whenever the developer specifies system.runtime.dll is one of the references.

Reply:
Ah, yeah that makes sense. We were relying on tfm name/description.